Change legacy wrap_dim to work with symint ==

wconstab opened the pull request:
- previously, sizes == vector<T>({0}) failed to hit SymInt::operator==, causing a the loop to bail out too early and make an invalid call to downstream maybe_wrap_dim helper

wconstab committed:
Change legacy wrap_dim to work with symint ==

- previously, sizes == vector<T>({0}) failed to hit SymInt::operator==,
  causing a the loop to bail out too early and make an invalid call
  to downstream maybe_wrap_dim helper

wconstab committed:
Add test case and update cat ref to match legacy_cat_wrap_dim

Chillee commented:
I'm not sure what's happening here? When does it not trigger now that it used to trigger?

wconstab committed:
Xfail preexisting functorch tests for newly added test case

- functorch tests for cat op used to pass,
- but i added a new opinfo sample which exposes a bug in functorch vmap/vjp

wconstab commented on aten/src/ATen/WrapDimUtils.h:
```diff
@@ -85,7 +85,7 @@ inline int64_t _legacy_cat_wrap_dim(
     int64_t dim,
     const std::vector<std::vector<T>>& tensor_sizes) {
   for (auto& sizes : tensor_sizes) {
-    if (sizes == std::vector<T>({0})) {
```
@Chillee here's what's going on:

sizes is {SymInt(0)}
std::vector<T>({0}) is actually {} for reasons i am still trying to understand.

so the previous code never did the 'continue' it wanted to do, but by explicitly checking .size() and sizes[0] it does.

wconstab committed:
Skip more tests
